metricsstorage/storage.go

```diff
@@ -0,0 +1,158 @@
+package metricsstorage
+
+import (
+	"context"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/microstorage"
+)
+
+const (
+	prometheusNamespace = "microstorage"
+
+	putActionName    = "put"
+	deleteActionName = "delete"
+	existsActionName = "exists"
+	listActionName   = "list"
+	searchActionName = "search"
+)
+
+var (
+	actionTotal = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: prometheusNamespace,
+			Name:      "action_total",
+			Help:      "Total number of storage actions performed.",
+		},
+		[]string{"action"},
+	)
+
+	errorTotal = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: prometheusNamespace,
+			Name:      "error_total",
+			Help:      "Total number of storage actions that have errored.",
+		},
+		[]string{"action"},
+	)
+
+	actionDuration = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: prometheusNamespace,
+			Name:      "action_duration_seconds",
+			Help:      "Duration of time to perform storage actions.",
+		},
+		[]string{"action"},
+	)
+)
+
+func init() {
+	prometheus.MustRegister(actionTotal)
+	prometheus.MustRegister(errorTotal)
+	prometheus.MustRegister(actionDuration)
+}
+
+type Config struct {
+	// Dependencies.
+	Underlying microstorage.Storage
+}
+
+func DefaultConfig() Config {
+	return Config{
+		// Dependencies.
```

Comment:
Just wondering about this comment inconsistency. IMO have it everywhere or nowhere.

Reply:
Sorting.